data-prepper-plugins/cloudwatch-logs/src/main/java/org/opensearch/dataprepper/plugins/sink/config/AwsConfig.java

```diff
@@ -0,0 +1,35 @@
+package org.opensearch.dataprepper.plugins.sink.config;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.Size;
+import software.amazon.awssdk.regions.Region;
+
+/**
+ * AwsConfig is based on the S3-Sink AwsAuthenticationOptions
+ * where the configuration allows the sink to fetch Aws credentials
+ * and resources.
+ */
+public class AwsConfig {
```

Comment:
Currently, AwsConfig is re-using code from AwsCredentialsOptions as AwsCredentialsOptions is not a general utils class. This would make the CloudWatchLogs plugin rely on a dependency from another plugin if we used AwsCredentialsOptions. I changed it to AwsConfig after talking to Krishna about this, we can discuss this further as well.